junit-platform-commons/src/main/java/org/junit/platform/commons/support/conversion/Converter.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.support.conversion;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import org.apiguardian.api.API;
+
+/**
+ * {@code Converter} is an abstraction that allows an input object to
+ * be converted to an instance of a different class.
+ *
+ * <p>Implementations are loaded via the {@link java.util.ServiceLoader} and must
+ * follow the service provider requirements. They should not make any assumptions
+ * regarding when they are instantiated or how often they are called. Since
+ * instances may potentially be cached and called from different threads, they
+ * should be thread-safe.
+ *
+ * <p>Extend {@link TypedConverter} if your implementation always converts
+ * from a given source type into a given target type and does not need access to
+ * the {@link ClassLoader} to perform the conversion.
+ *
+ * @since 1.13
+ * @see ConversionSupport
+ * @see TypedConverter
+ */
+@API(status = EXPERIMENTAL, since = "1.13")
+public interface Converter {
```

Comment:
Didn't we decide to make this `generic` as in `Converter<S, T>`?

Reply:
I guess for that to make sense, we'd have to parameterize `TypeDescriptor` as well.

@scordio @sbrannen WDYT?

Reply:
I had the same in mind but wasn't entirely confident about this direction yesterday, especially before refactoring `StringToObjectConversion`.

But now the refactoring is done so I can sketch it and see what happens 🙂

Reply:
Started in 8b78923, there is still work to do.

@marcphilipp @sbrannen could you please check if this is the direction you have in mind?